diff --git a/cmake/onnxruntime.cmake b/cmake/onnxruntime.cmake
index e2871b5..12c085b 100644
--- a/cmake/onnxruntime.cmake
+++ b/cmake/onnxruntime.cmake
@@ -143,7 +143,7 @@ if(SHERPA_ONNX_USE_PRE_INSTALLED_ONNXRUNTIME_IF_AVAILABLE)
     if(APPLE)
       set(location_onnxruntime_lib $ENV{SHERPA_ONNXRUNTIME_LIB_DIR}/libonnxruntime.dylib)
     elseif(WIN32)
-      if(SHERPA_ONNX_ENABLE_GPU)
+      if(SHERPA_ONNX_ENABLE_GPU OR SHERPA_ONNX_ENABLE_DIRECTML)
         set(location_onnxruntime_lib $ENV{SHERPA_ONNXRUNTIME_LIB_DIR}/onnxruntime.dll)
         set(location_onnxruntime_lib2 $ENV{SHERPA_ONNXRUNTIME_LIB_DIR}/onnxruntime.lib)
       else()
